updated docs for the graph -- full walkthrough tutorial

alinobrasil opened the pull request:
Here's an updated version of the docs that takes a dev from start to finish in creating a subgraph, deploying & publishing it and finally querying it. This includes screenshots specific to deploying on Etherlink.

timothymcmackin reviewed (changes requested):
Thanks for the submission! Some review comments:

- For consistency with the rest of the site, please change `dapp` to `dApp`.
- Overuse of headings for each step: please use numbered steps instead of making every step a heading, like "Create a subgraph on Subgraph Studio," "Install the Graph CLI⁠," and "Initialize your Subgraph⁠."
- You're removed the prerequisite about verifying the contract and reduced it to "imply have your contract verified on the block explorer." I think we should at least provide the link to the topic on verifying the contract as in the current prereq.
- Heading capitalization: some headings capitalize every major word and others don't. For consistency with the rest of the docs, please capitalize the first letter of the heading and then only capitalize proper nouns like The Graph and Etherlink (sentence case).
- What's GRT? The instructions mention it but don't define it.
- In some places you use `TheGraph` with no space and in other places it's `The Graph` with a space -- which is it?
- Remove pricing information -- this isn't the place to sell a service and we have no way of keeping it up to date as The Graph prices change.
- Why is it important that "The Graph's smart contracts are all on Arbitrum One?"

alinobrasil commented:
Thanks for the review. I've addressed these points:

> * For consistency with the rest of the site, please change `dapp` to `dApp`. => Done
> * Overuse of headings for each step: please use numbered steps instead of making every step a heading, like "Create a subgraph on Subgraph Studio," "Install the Graph CLI⁠," and "Initialize your Subgraph⁠." => Done. Using numbered list. 
> * You're removed the prerequisite about verifying the contract and reduced it to "imply have your contract verified on the block explorer." I think we should at least provide the link to the topic on verifying the contract as in the current prereq. => Done. Added link at the end of the first part "Initialize your subgraph". (Just before "Deploy & publish") 
> * Heading capitalization: some headings capitalize every major word and others don't. For consistency with the rest of the docs, please capitalize the first letter of the heading and then only capitalize proper nouns like The Graph and Etherlink (sentence case). ==> Done
> * What's GRT? The instructions mention it but don't define it. ==> Done. Defined as The Graph's token. 
> * In some places you use `TheGraph` with no space and in other places it's `The Graph` with a space -- which is it? ==> The Graph. This has been fixed in the title.
> * Remove pricing information -- this isn't the place to sell a service and we have no way of keeping it up to date as The Graph prices change. ==> Done
> * Why is it important that "The Graph's smart contracts are all on Arbitrum One?" ==> We've seen developers often get confused about where subgraphs are deployed, mixing up the chain being **indexed** vs the chain where The Graph's subgraphs are **deployed**.

timothymcmackin reviewed (approved):
Thanks! I'm approving and am going to go ahead and merge.
